.gitattributes

```diff
@@ -1 +0,0 @@
-* -text
```

Comment:
Please, restore that file. While we can enforce line endings for our code that is not the case for third-party code. Without that setting we have to inform every windows user (https://github.com/SmingHub/Sming/wiki/Windows-Quickstart#configure-git-for-windows-environment) how to use git and how to set the line endings in order for the patching to work there.

Reply:
Can you suggest an alternative solution to the problem (other than not using Windows!) ?

Reply:
In order to track changes through commits properly, every single commit needs to be consistently checked in with autocrlf behaviour. To fix a bunch of commits I used this:

```
git filter-branch --tree-filter "git add --renormalize ." develop..HEAD
```

But that doesn't work with the current setting in `.gitattributes`, so I had to rebase to get rid of that first before running the filter.

Could we perhaps amend the filter so it specifies which files it should/shouldn't apply to?

Reply:
Our code should use Linux line endings. That can be enforced with running dos2unix. 
When a contributor has added new code and his IDE has changed the line endings he should run dos2unix on his local branch, commit  the dos2unix change and only then rebase. 

> To fix a bunch of commits I used this:

Since I am squash-merging the PRs it does not really matter if the dos2unix was the first commit in the PR, the middle one or the last one.

Reply:
>
> While we can enforce line endings for our code that is not the case for third-party code.

Wouldn't third-party code be in a submodule anyway? What else would qualify as 'third-party' code and might require different line endings?

>
> Without that setting we have to inform every windows user (https://github.com/SmingHub/Sming/wiki/Windows-Quickstart#configure-git-for-windows-environment) how to use git and how to set the line endings in order for the patching to work there.

Patches are applied using `--ignore-whitespace --whitespace=nowarn` - is that not sufficient? I'm trying to understand the problem which .gitattributes (#1421) is supposed to fix?

>
> Since I am squash-merging the PRs it does not really matter if the dos2unix was the first commit in the PR, the middle one or the last one.

No, but with complex PR's (like the docs update) if there aren't consistent line endings through every commit it makes tracking changes and rebasing almost impossible.

What I'm trying to pin down is how to stop this problem occurring in the first place, without having to fix it afterwards and complicate the dev. process un-necessarily.

Reply:
> but with complex PR's (like the docs update) if there aren't consistent line endings 

Ok,  If that makes life easier let's remove .gitattributes.